release/tar/linux/opensearch-tar-build.sh

```diff
@@ -0,0 +1,175 @@
+            ;;
+        a)
+            ARCHITECTURE=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$PLATFORM" ]; then
+    PLATFORM="linux"
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$ARCHITECTURE" ]; then
+    ARCHITECTURE="x64"
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "error: You must specify the manifest file"
+    usage
+    exit 1
+fi
+
+if [ ! -f $MANIFEST_FILE ]; then
+    echo "error: The given file doesnot exist. Please check the specified path"
+    usage
+    exit 1
+fi
+
+SCHEMA_VERSION=`yq eval '.manifest.schema-version' $MANIFEST_FILE`
+echo $SCHEMA_VERSION
+
+if [ $SCHEMA_VERSION != 1.0 ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
+    usage
+    exit 1
+fi
+
+REPO_ROOT=`git rev-parse --show-toplevel`
+VERSION=`yq eval '.version' $MANIFEST_FILE`
+WORKING_DIR="opensearch-working-directory"
+OPENSEARCH_CORE_URL=`yq eval '.products.opensearch.opensearch-min' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+OPENSEARCH_PLUGINS_URLS=`yq eval '.products.opensearch.plugins' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+LIBS=`yq eval '.products.opensearch.libs' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+
+# # Download plugins to local 
+mkdir -p ${WORKING_DIR}/plugins_downloads
+IFS=$'\n'
+cd ${WORKING_DIR}/plugins_downloads
+for plugin in ${OPENSEARCH_PLUGINS_URLS}; do
+  aws s3 cp $plugin .
+done
+
+# # Download k-NN lib
+aws s3 cp $LIBS .
+unzip opensearch-knnlib-*-$PLATFORM-$ARCHITECTURE.zip
+
+cd ..
+
+# Download OpenSearch core TAR
+wget -q ${OPENSEARCH_CORE_URL} -O opensearch.tar.gz
+tar -xzf opensearch.tar.gz
+rm opensearch.tar.gz
+
+# Install plugins to OpenSearch
+cd opensearch-*
+for plugins in $OPENSEARCH_PLUGINS_URLS; do
+    plugin=`basename $plugins`
+  ./bin/opensearch-plugin install --batch file:../plugins_downloads/$plugin
+done
+
+echo "Plugins installed are:\n"
+ls -tlr plugins
+
+# Setup Performance Analyzer Agent
+cp -r plugins/opensearch-performance-analyzer/performance-analyzer-rca .
+chmod -R 755 performance-analyzer-rca
+mv bin/opensearch-performance-analyzer/performance-analyzer-agent-cli ./bin
+rm -rf ./bin/opensearch-performance-analyzer
+
+
+# Make sure the data folder exists and is writable
+if [ ! -d "data" ]; then 
+    mkdir data
```

Comment:
Consider rewriting as `mkdir data 2>/dev/null` to so we can avoid the comment/if statement

Reply:
Offline recap: based on the usage of `set -e` at the top of the file this suggestion could introduce a hard to detect bug! Might be useful to have a script style overview where we figure out what styles make the most sense for us as a team - I'll drive setting this up

Reply:
`mkdir -p` does not generate an error if the directory already exists. Use that and get rid of the `if`.

Reply:
done